### DISCUSSION TITLE
zh/questions

### DISCUSSION THREAD

**Discussion by giscus** [tzfun/vue-web-terminal#109] [Announcements]
# zh/questions

一个轻量、功能强大的命令式网页仿真终端插件

http://www.beifengtz.com/zh/questions.html

**Comment by shuiYe-704265085** [comment]
我编写了一个简单的ssh demo，前端只做显示，所有内容由后端建立ssh连接返回给前端，有几个问题
1. 前端的context和context-suffix设置为“”空字符串，但是后端返回命令提示符后还是会换行。比如后端返回~$，有没有办法让光标显示在$后面不换行
2. 我发现执行命令后会有一行是单独显示命令的，应该是因为第一点我把context设置成""
备注：全程使用TerminalApi.appendMessage进行追加

**Reply by tzfun** [reply]
> 我编写了一个简单的ssh demo，前端只做显示，所有内容由后端建立ssh连接返回给前端，有几个问题
> 
> 1. 前端的context和context-suffix设置为“”空字符串，但是后端返回命令提示符后还是会换行。比如后端返回~$，有没有办法让光标显示在$后面不换行
> 2. 我发现执行命令后会有一行是单独显示命令的，应该是因为第一点我把context设置成""
>    备注：全程使用TerminalApi.appendMessage进行追加

其实这两个是一个问题

后端返回 `~$` 表示需要输入新的指令，这种模式并不适合本插件，建议切换为command模式，也就是后端发送指令，ssh返回结果，后端将结果处理之后发送给前端显示。

你目前使用的ssh连接模式是会实时回显远程主机的终端内容（名字忘记了），这种模式下比较适合做ssh客户端，也就是你平时在ssh连接终端上看到的所有内容都是远程主机返回的，包括你的键入反馈、标题、光标等，该模式通常需要设定终端的行数、列数、光标位置等信息，它的实现机制与本插件不同，并不适合。

**Reply by shuiYe-704265085** [reply]
好的，谢谢，因为之前xterm可以直接渲染，想着会不会也有什么配置，现在后端伪终端改一下模式不返回命令提示符就正常了
